bindings/go/src/ucx/request.go

```diff
@@ -29,7 +29,7 @@ func (p *UcpRequestParams) SetMemType(memType UcsMemoryType) *UcpRequestParams {
 	return p
 }
 
-func (p *C.ucp_request_param_t) SetMemType(params *UcpRequestParams) {
+func SetMemType(params *UcpRequestParams, p *C.ucp_request_param_t) {
```

Comment:
Do I understand correctly that this is some kind of casting function, which casts Go structure to C one?

I mean is that function for internal use only? Or is it supposed to be used by customers? If latter, then this change would break API compatibility.

Reply:
The function sets field to a C-structure based on values of Go-structure passed as a parameter.
Originally, it was written in a way that allows to have the following code:
```go
func foo(goParams *UcpRequestParams, cParams *C.ucp_request_param_t) {
    ...
    cParams.SetMemType(goRequestParams)
    ...
}
```
However, go doesn't allow to extend an existing type (`C.ucp_request_param_t`) in another package (`ucx`).
The older go version don't raise an error, unlike the newer ones.
Yes this change would break compatibility. But the code still needs to be modified so that it can be compiled.

Reply:
does it mean we need the same fix when setting all other fields of UcpRequestParams ?

Reply:
How do you think, does it worth it to add some compile-time logic to leave the old version of function on place for old versions of go?

Reply:
AFAIK this is an internal function , let's see how we can "hide" it

Reply:
Made it a private.